src/vs/platform/workspace/common/workspace.ts

```diff
@@ -75,6 +80,11 @@ export const enum WorkbenchState {
 	WORKSPACE
 }
 
+export interface IWorkspaceFoldersWillChangeEvent {
+	updateWorkspaceTrustState(promise: Promise<void>): void;
```

Comment:
Simplify this event as follows: 

```
export interface IWorkspaceFoldersWillChangeEvent {
 	join(promise: Promise<void>, changes: IWorkspaceFoldersChangeEvent): void;
}
```

Reply:
My bad, `join` cannot have `changes` arg - since it is called by the event callback handler. I think rename should be enough as follows:

```
export interface IWorkspaceFoldersWillChangeEvent {
     join(promise: Promise<void>): void;
     readonly changes: IWorkspaceFoldersChangeEvent;
}
```